runtime/drivers/duckdb/catalog.go

```diff
@@ -30,6 +35,12 @@ func (c *connection) FindObject(ctx context.Context, instanceID string, name str
 }
 
 func (c *connection) CreateObject(ctx context.Context, instanceID string, obj *drivers.CatalogObject) error {
+	// safeguard to make sure duplicates are not created
+	_, found := c.FindObject(ctx, instanceID, obj.Name)
+	if found {
+		return errors.New(fmt.Sprintf("duplicate key : %s", obj.Name))
+	}
+
```

Comment:
I actually think you could just add a duckdb migration with an index:

```
CREATE UNIQUE INDEX lower_name_unique_idx ON catalog (lower(name));
```

Then you also don't have possible race conditions (though very unlikely given that this is local)